bokeh/resources.py

```diff
@@ -671,50 +702,47 @@ def _get_cdn_urls(version=None, minified=True, legacy=False):
     # check the 'dev' fingerprint
     container = dev_container if _DEV_PAT.match(version) else rel_container
 
-    def mk_filename(comp, kind):
+    def mk_filename(comp: str, kind: Kind) -> str:
         return f"{comp}-{version}{_legacy}{_minified}.{kind}"
 
-    def mk_url(comp, kind):
+    def mk_url(comp: str, kind: Kind) -> str:
         return f"{base_url}/{container}/" + mk_filename(comp, kind)
 
-    result = {
-        "urls": lambda components, kind: [mk_url(component, kind) for component in components],
-        "messages": [],
-        "hashes" : None
-    }
+    result = Urls(urls=lambda components, kind: [mk_url(component, kind) for component in components])
 
     if len(__version__.split("-")) > 1:
-        result["messages"].append(
-            {
-                "type": "warn",
-                "text": (
-                    "Requesting CDN BokehJS version '%s' from Bokeh development version '%s'. "
-                    "This configuration is unsupported and may not work!" % (version, __version__)
-                ),
-            }
-        )
-
-    if is_full_release(version):
+        result.messages.append(Message(
```

Comment:
"Message" is awfully generic. Can we make this type more specific? AFAIK this is only / mostly used to publish messages to notebook output cells but it might be OK to denote it as a "JS runtime message" in some way